node/web/api/booksApiController.js

```diff
@@ -26,5 +25,82 @@ module.exports.factory = function (router, repo, exceptions) {
         });
     });
 
+    router.post('/api/product', function (req, res) {
+      var count = "";
+      var bookTitle = "";
+      repo.get(req.query.product, function (err, book) {
+            if (err) {
+                exceptions.throwException(err);
+                res.status(400).end();
+                return;
+            }
+            var email = "";
+            console.log("Book title: " + book.title);
+            if(!req.cookies.email)
+            {
+
+              email = "Guest";
+            }
+            else {
+              {
+                email = req.cookies.email;
+              }
+            }
+            ordersRepo.updateCart(email,book, function (err, user){
+              if (err){
+                res.status(400).end();
+                return;
+              }
+              });
+              ordersRepo.getCount(email, function (err,doc) {
```

Comment:
By executing this synchronously, you introduce a race condition. The query being executed by `getCount` can actually fire before the cart update completes, in which case your cart count would not return what you expect. To remove the race condition:

```JavaScript
ordersRepo.updateCart(email, book, function (err, user) {
    if (err){
        res.status(400).end();
        return;
    }

    ordersRepo.getCount(email, function (err,doc) {
        if (err) {
            res.status(400).end();
            return;
        }

        count = doc.total_quantity;
        res.send(String(count));
    });    
});
```